ts/src/bingx.ts

```diff
@@ -0,0 +1,2317 @@
+            result.push (this.parseMarket (markets[i]));
+        }
+        return result;
+    }
+
+    async fetchSwapMarkets (params) {
+        const response = await this.swapV2PublicGetQuoteContracts (params);
+        //
+        //    {
+        //        "code": 0,
+        //        "msg": "",
+        //        "data": [
+        //          {
+        //            "contractId": "100",
+        //            "symbol": "BTC-USDT",
+        //            "size": "0.0001",
+        //            "quantityPrecision": 4,
+        //            "pricePrecision": 1,
+        //            "feeRate": 0.0005,
+        //            "tradeMinLimit": 1,
+        //            "maxLongLeverage": 150,
+        //            "maxShortLeverage": 150,
+        //            "currency": "USDT",
+        //            "asset": "BTC",
+        //            "status": 1
+        //          },
+        //        ]
+        //    }
+        //
+        const result = [];
+        const markets = this.safeValue (response, 'data');
+        for (let i = 0; i < markets.length; i++) {
+            result.push (this.parseMarket (markets[i]));
+        }
+        return result;
+    }
+
+    parseMarket (market) {
+        const id = this.safeString (market, 'symbol');
+        const symbolParts = id.split ('-');
+        const baseId = symbolParts[0];
+        const quoteId = symbolParts[1];
+        const base = this.safeCurrencyCode (baseId);
+        const quote = this.safeCurrencyCode (quoteId);
+        const currency = this.safeString (market, 'currency');
+        const settle = this.safeCurrencyCode (currency);
+        let pricePrecision = this.safeInteger (market, 'pricePrecision');
+        if (pricePrecision === undefined) {
+            pricePrecision = this.precisionFromString (this.safeString (market, 'tickSize'));
+        }
+        let quantityPrecision = this.safeInteger (market, 'quantityPrecision');
+        if (quantityPrecision === undefined) {
+            quantityPrecision = this.precisionFromString (this.safeString (market, 'stepSize'));
+        }
+        const type = (settle !== undefined) ? 'swap' : 'spot';
+        const spot = type === 'spot';
+        const swap = type === 'swap';
+        let symbol = base + '/' + quote;
+        if (settle !== undefined) {
+            symbol += ':' + settle;
+        }
+        const contractSize = this.safeNumber (market, 'size');
+        const entry = {
+            'id': id,
+            'symbol': symbol,
+            'base': base,
+            'quote': quote,
+            'settle': settle,
+            'baseId': baseId,
+            'quoteId': quoteId,
+            'settleId': currency,
+            'type': type,
+            'spot': spot,
+            'margin': false,
+            'swap': swap,
+            'future': false,
+            'option': false,
+            'active': this.safeString (market, 'status') === '1' ? true : false,
```

Comment:
`this.safeString (market, 'status') === '1'` already returns a boolean no need for the conditional here